File: components/core/src/main/java/org/storm/core/input/action/SimpleActionManager.java
package org.storm.core.input.action;

import java.util.HashMap;

/**
 * Simple implementation of an ActionManager.
 */
public class SimpleActionManager extends ActionManager {

  public SimpleActionManager() {
    super(new HashMap<>());
  }

  @Override
  public void add(String action) {
    this.state.put(action, false);
  }

  @Override
  public void remove(String action) {
    this.state.remove(action);
  }

  @Override
  public void startUsing(String action) {
    this.state.put(action, true);
  }

  @Override
  public void stopUsing(String action) {
    this.state.put(action, false);
  }

}

File: components/core/src/main/kotlin/org/storm/core/input/action/SimpleActionManager.kt
package org.storm.core.input.action

/**
 * Simple implementation of an ActionManager.
 */
class SimpleActionManager : ActionManager(HashMap()) {
  override fun add(action: String) {
    this.state[action] = false
  }

  override fun remove(action: String) {
    state.remove(action)
  }

  override fun startUsing(action: String) {
    state[action] = true
  }

  override fun stopUsing(action: String) {
    state[action] = false
  }
}
